src/app/shared/services/user-session.service.ts

```diff
@@ -17,35 +18,46 @@ export interface UserSession {
 export class UserSessionService implements OnDestroy {
 
   session$ = new BehaviorSubject<UserSession|undefined>(undefined)
+  userProfileError$ = new Subject<void>()
+  watchedGroup$ = this.session$.pipe(map(session => session?.watchedGroup), distinctUntilChanged())
 
   /** currently-connected user profile, temporary or not, excluding transient (undefined) states */
-  user$ = this.session$.pipe(
+  userProfile$ = this.session$.pipe(
     filter(isNotUndefined),
     map(session => session.user),
-    distinctUntilChanged((u1, u2) => u1.groupId === u2.groupId)
   );
 
+  /** triggered when the user identity changes, which happens when auth token is invalidated */
+  userChanged$ = this.userProfile$.pipe(distinctUntilChanged((u1, u2) => u1.groupId === u2.groupId), mapTo(undefined), skip(1))
+
   private subscription?: Subscription;
+  private refresh$ = new Subject<void>()
```

Comment:
For a user-session service, "refresh$" is not clear enough (could be many things to refresh). 
`userProfileRefresh$`  ?

Reply:
Fixed